### PR TITLE
Feature/gemeni cache

### LINKED ISSUE #76: [Feature] Gemini with lRU catch - upload api
- [ ] gemini 放入LRU catch, 完成 upload api

### LINKED ISSUE #77: [Feature] Gemini with lRU catch - process api
- [ ] gemini 放入LRU catch, 完成 progress api

### LINKED ISSUE #78: [Feature] Gemini with lRU catch - result api
gemini 放入LRU catch, 完成 result api

### PR DESCRIPTION
**Pull request by TinyMurky**
### DEVELOP

- feat: Gemini 套入 Lru cache 完成 upload, process status, result的三個步驟
  
#### Related Issues

- [X] Issue Number:
    - #76 
    - #77 
    - #78 

#### Checklist

- [x] Used `@` in import paths.
- [x] Verified naming convention compliance ([Naming Convention Guidelines](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md)).
- [x] Coding style verification: checked
- [x] new Library: 0
- [x] new Class / Component: 0
- [x] new loop: 0
- [x] new recursive: 0
- [x] high risk: 0
- [x] new sql: 0

#### UML Diagrams

- None

#### Additional Notes

- Ensured all links are up-to-date.
- Conducted thorough testing on responsive design changes.
- Documented any notable considerations or edge cases addressed.

### CODE REVIEW

File: src/api/gemini/gemini.module.ts
Comment (Luphia) [inline src/api/gemini/gemini.module.ts:1]: - use '@' for import path

File: src/api/gemini/gemini.service.spec.ts
Comment (Luphia) [inline src/api/gemini/gemini.service.spec.ts:1]: - use '@' for import path

File: src/api/gemini/gemini.service.ts
Comment (Luphia) [inline src/api/gemini/gemini.service.ts:1]: - Annotation should be comply with the annotation format guidelines

File: src/api/gemini/gemini.controller.ts
Comment (Luphia) [inline src/api/gemini/gemini.controller.ts:1]: - Annotation should be comply with the annotation format guidelines

**Review by Luphia — CHANGES_REQUESTED** [review]

File: src/api/gemini/gemini.controller.ts
Comment (TinyMurky) [reply src/api/gemini/gemini.controller.ts:1]: 修改完成！

File: src/api/gemini/gemini.module.ts
Comment (TinyMurky) [reply src/api/gemini/gemini.module.ts:1]: 修改完成！

File: src/api/gemini/gemini.service.spec.ts
Comment (TinyMurky) [reply src/api/gemini/gemini.service.spec.ts:1]: 修改完成！

File: src/api/gemini/gemini.service.ts
Comment (TinyMurky) [reply src/api/gemini/gemini.service.ts:1]: 修改完成！

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**